### PR TITLE
Feature/fix rsem

### PR DESCRIPTION
**Pull request by jerowe**
* [ ] I have read the guidelines above.
* [ ] This PR adds a new recipe.
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

@bgruening , can you take a look at this? There are a lot of things I ad to change with this recipe.

### CODE REVIEW

File: recipes/rsem/meta.yaml [hunk 1]
```diff
@@ -1,27 +1,54 @@
...
-  version: 1.2.28
+  name: {{ name }} 
+  version: {{ version }}
+
 build:
   rpaths:
     - lib/R/lib/
     - lib/
-  number: 0
+  number: 2
```
Comment (bgruening) [inline recipes/rsem/meta.yaml]: build number 1?

File: recipes/rsem/meta.yaml [hunk 2]
```diff
@@ -1,27 +1,54 @@
...
+    - htslib 1.3
+    - zlib
     - r
+    - perl
+    - bioconductor-biobase
+    - perl-module-build
   run:
+    - libgcc
+    - zlib
+    - boost
```
Comment (bgruening) [inline recipes/rsem/meta.yaml]: please pin boost and change the build-string

File: recipes/rsem/build.sh [hunk 3]
```diff
@@ -34,6 +37,55 @@ rsem-tbam2gbam \
...
+export LD_LIBRARY_PATH="${PREFIX}/lib"
+
+export LDFLAGS="-L${PREFIX}/lib"
+export CPPFLAGS="-I${PREFIX}/include"
+
+########################################################
+### Prepare Samtools/Htslib 
+########################################################
+
+sed -i.bak 's/^CPPFLAGS =$//g' samtools-1.3/Makefile
```
Comment (bgruening) [inline recipes/rsem/build.sh]: If you depend on samtools why do you compile this version as well?

File: recipes/rsem/build.sh [hunk 4]
```diff
@@ -34,6 +37,55 @@ rsem-tbam2gbam \
...
+
+# Remove rdynamic which can cause build issues on OSX
+# https://sourceforge.net/p/samtools/mailman/message/34699333/
+sed -i.bak 's/ -rdynamic//g' samtools-1.3/Makefile
+sed -i.bak 's/ -rdynamic//g' samtools-1.3/htslib-1.3/configure
+
+########################################################
+### Build htslib 
+########################################################
+cd samtools-1.3/htslib-1.3
```
Comment (bgruening) [inline recipes/rsem/build.sh]: Same here, the libs and includes should be in your $PREFIX as you depend on it, already.

File: recipes/rsem/Makefile.new [hunk 5]
```diff
@@ -0,0 +1,158 @@
+CC = g++
+CFLAGS = -Wall -c -I.
```
Comment (bgruening) [inline recipes/rsem/Makefile.new]: I guess both Makefiles, the new and the orig, are not needed in this PR as you have the patch?

File: recipes/rsem/build.sh (hunk 3 above)
Comment (jerowe) [inline recipes/rsem/build.sh]: RSEM needs to compile against specific samtools libraries, and for my first iteration I just wanted to get it working. I'm going to rework the `Makefile` it so I don't have to do this.

Comment (jerowe) [inline recipes/rsem/build.sh]: Upon further inspection rsem requires some header files that are not in the conda package. I run make to get the headers, but use the bioconda package for the actual installed versions.

Comment (bgruening) [inline recipes/rsem/build.sh]: That would be great! Thanks!

**Comment by jerowe** [comment]
Merge?

**Comment by bgruening** [comment]
Merge if you like! :)

**Pull request merged**